src/App.jsx

```diff
@@ -14,70 +16,68 @@ export const goods = [
   'Garlic',
 ];
 
-export const App = () => (
-  <main className="section container">
-    <h1 className="title is-flex is-align-items-center">No goods selected</h1>
+export const App = () => {
+  const [selectedGood, setSelectedGood] = useState('Jam');
 
-    <h1 className="title is-flex is-align-items-center">
-      Jam is selected
+  return (
+    <main className="section container">
+      <h1 className="title is-flex is-align-items-center">
+        {selectedGood
+          ? (
+            <>
+              {`${selectedGood} is selected`}
 
-      <button
-        data-cy="ClearButton"
-        type="button"
-        className="delete ml-3"
-      />
-    </h1>
+              <button
+                data-cy="ClearButton"
+                type="button"
+                className="delete ml-3"
+                onClick={() => setSelectedGood('')}
+              />
+            </>
+          )
+          : (
+            'No goods selected'
+          )
+        }
+      </h1>
 
-    <table className="table">
-      <tbody>
-        <tr data-cy="Good">
-          <td>
-            <button
-              data-cy="AddButton"
-              type="button"
-              className="button"
-            >
-              +
-            </button>
-          </td>
+      <table className="table">
+        <tbody>
+          {goods.map((good) => {
+            const isGoodSelected = good === selectedGood;
 
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Dumplings
-          </td>
-        </tr>
+            return (
+              <tr
+                data-cy="Good"
+                key={good}
+                className={classNames({
+                  'has-background-success-light': isGoodSelected,
+                })}
+              >
+                <td>
+                  <button
+                    data-cy={isGoodSelected
+                      ? 'RemoveButton' : 'AddButton'}
+                    type="button"
+                    className={classNames('button', {
+                      'is-info': isGoodSelected,
+                    })}
+                    onClick={() => setSelectedGood(isGoodSelected
+                      ? ''
+                      : good)}
+                  >
+                    {isGoodSelected ? '-' : '+'}
+                  </button>
```

Comment:
Now everything is written correctly, but such code is difficult to read and maintain. It can be broken down into smaller parts:
```js
const isGoodSelected = good === selectedGood;
const buttonDataCy = isGoodSelected ? 'RemoveButton' : 'AddButton';
const buttonValue = isGoodSelected ? '-' : '+';
const buttonClassName = classNames('button', {
   'is-info': isGoodSelected,
 });
const handleGoodSelection = () => setSelectedGood(isGoodSelected ? '' : good);
...
<button
  data-cy={buttonDataCy}
  type="button"
  className={buttonClassName}
  onClick={handleGoodSelection}
>
  {buttonValue}
</button>
```
Such code takes up more space, but it is easier to read and change if necessary.